keras/layers/convolutional_recurrent.py

```diff
@@ -1,172 +1,585 @@
 # -*- coding: utf-8 -*-
+"""Convolutional-recurrent layers.
+"""
 from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
 
 from .. import backend as K
 from .. import activations
 from .. import initializers
 from .. import regularizers
 from .. import constraints
-from .recurrent import Recurrent
+from .recurrent import _generate_dropout_mask
 
 import numpy as np
-from ..engine import InputSpec
+from ..engine import InputSpec, Layer
 from ..utils import conv_utils
 from ..legacy import interfaces
+from ..legacy.layers import Recurrent, ConvRecurrent2D
+from ..utils.generic_utils import has_arg
 
 
-class ConvRecurrent2D(Recurrent):
-    """Abstract base class for convolutional recurrent layers.
-
-    Do not use in a model -- it's not a functional layer!
+class ConvRNN2D(Layer):
```

Comment:
I believe you should be to remove a lot of redundant code by subclassing `RNN`. There are lots of shared methods.